components/org.wso2.openbanking.cds.consent.extensions/src/main/java/org/wso2/openbanking/cds/consent/extensions/admin/impl/CDSConsentAdminHandler.java

```diff
@@ -521,6 +538,92 @@ private boolean isActionByPrimaryUser(DetailedConsentResource detailedConsentRes
+        ArrayList<String> userIDs = (ArrayList<String>) consentAdminData.getQueryParams()
+                .get(CDSConsentExtensionConstants.USER_IDS_QUERY_PARAM_NAME);
+        if (userIDs == null || userIDs.isEmpty()) {
+            // Customer care officer scenario
+            consentData.forEach(consent -> {
+                JSONObject consentObject = (JSONObject) consent;
+                consentObject.put(CDSConsentExtensionConstants.CAN_REVOKE, true);
+            });
+        } else {
+            String userId = userIDs.get(0);
+            consentData.forEach(consent -> {
+                JSONObject consentObject = (JSONObject) consent;
+                consentObject.put(CDSConsentExtensionConstants.CAN_REVOKE, canRevokeByBNR(consentAdminData, userId));
+            });
+        }
+    }
+
+    private boolean canRevokeByBNR(DetailedConsentResource detailedConsentResource, String userId) {
+
+        Set<String> activeAccountIDs = new HashSet<>();
+        for (ConsentMappingResource consentMappingResource : detailedConsentResource.getConsentMappingResources()) {
+            if (consentMappingResource.getMappingStatus().equals(CDSConsentExtensionConstants.ACTIVE_STATUS)) {
+                activeAccountIDs.add(consentMappingResource.getAccountID());
+            }
+        }
+
+        return canRevokeByBNR(activeAccountIDs, userId);
+    }
+
+    private boolean canRevokeByBNR(ConsentAdminData consentAdminData, String userId) {
+
+        Set<String> activeAccountIDs = new HashSet<>();
+
+        for (Object item : (JSONArray) consentAdminData.getResponsePayload().
+                get(CDSConsentExtensionConstants.DATA)) {
+            JSONObject itemJSONObject = (JSONObject) item;
+            JSONArray consentMappingResourcesArray = (JSONArray) itemJSONObject.
+                    get(CDSConsentExtensionConstants.CONSENT_MAPPING_RESOURCES);
+            for (Object consentMappingResource : consentMappingResourcesArray) {
+                JSONObject consentMappingResourceObject = (JSONObject) consentMappingResource;
+                if (consentMappingResourceObject.getAsString(CDSConsentExtensionConstants.MAPPING_STATUS).
+                        equals(CDSConsentExtensionConstants.ACTIVE_STATUS)) {
+                    activeAccountIDs.add(consentMappingResourceObject.getAsString(
+                            CDSConsentExtensionConstants.ACCOUNT_ID));
+                }
+            }
+        }
+        return canRevokeByBNR(activeAccountIDs, userId);
+    }
+
+    private boolean canRevokeByBNR(Set<String> activeAccountIDs, String userId) {
+
+        if (activeAccountIDs.isEmpty()) {
+            return true;
+        }
+
+        AccountMetadataServiceImpl accountMetadataService = AccountMetadataServiceImpl.getInstance();
+
+        /*
+         * For all the active accounts, if the user has AUTHORIZE or REVOKE permission for BNR, then the user can
```

Comment:
User can revoke the consent only if he has authorize permission.

Reply:
Fixed with https://github.com/wso2/reference-implementation-consumerdatastandards-au/pull/66/commits/e7670fae0d8a1d0bcc940f03eed1cf98bab04973